src/recordlinker/schemas/pii.py

```diff
@@ -234,36 +234,32 @@ class Telecom(StrippedBaseModel):
     system: typing.Optional[str] = None
     use: typing.Optional[str] = None
 
-    def phone(self) -> str | None:
-        """
-        Return the phone number from the telecom record.
-        """
-        if self.system != "phone":
-            return None
-        # normalize the number to include just the 10 digits
-        return re.sub(r"\D", "", self.value).strip()[:10]
-
-    def email(self) -> str | None:
-        """
-        Return the email address from the telecom record.
-        """
-        if self.system != "email":
-            return None
-        return self.value.lower().strip()
-
-    @classmethod
-    @functools.lru_cache()
-    def get_system_handlers(cls) -> dict[str, str]:
-        """
-        Return a dictionary of system handlers for the Telecom class where the keys
-        are the system values and the values are the method names to call.
-
-        """
-        return {
-            name: name
-            for name, method in cls.__dict__.items()
-            if callable(method) and not name.startswith("_")
-        }
+    @pydantic.model_validator(mode="after")
+    def validate_and_normalize_telecom(self):
+        """
+        Validate and normalize the telecom record.
+        """
+        # If telecom.system = "email", set telecom.value to lowercase
+        #
+        if self.system == "email":
+            self.value = self.value.strip().lower()
+        # If telecom.system = "phone", normalize the number
+        elif self.system == "phone":
+            try:
+                # Attempt to parse with country code
+                if self.value.startswith("+"):
+                    parsed_number = phonenumbers.parse(self.value)
```

Comment:
I don't think we're doing anything with the `parsed_number` variable.  Shouldn't we be assigning self.value to it?